### ISSUE TITLE
Bug: GPT-OSS TRT engine ignores YaRN rope_scaling and sliding-window layer_types, garbling prompts longer than 128 tokens

### ISSUE THREAD

**Issue by chaofengw-nv** [NVIDIA/TensorRT-Model-Connect#399]
## Summary

The GPT-OSS TensorRT engine builders silently drop two architectural features of the model, so built engines diverge from the HF reference and generate garbage for any prompt longer than the 128-token sliding window:

1. **YaRN RoPE scaling is never applied.** `openai/gpt-oss-20b`'s `config.json` serializes the scaling dict under `rope_scaling`, but both `families/gpt_oss/plugin.py` (`build_engine`) and `families/gpt_oss/tp_builder.py` (`_make_rope_tables`) read only `config.raw["rope_parameters"]`. The lookup misses, `rope_type` falls back to `"default"`, and the engine gets plain RoPE tables — no frequency interpolation (factor 32) and no attention factor (`0.1*ln(32)+1 ≈ 1.3466`). The existing `make_yarn_rope_table_half_dim` also always truncates the correction range and never applies the attention factor, while GPT-OSS ships `"truncate": false`.
2. **Sliding-window attention is not implemented.** GPT-OSS alternates `sliding_attention` (window 128) and `full_attention` layers (`layer_types` in config.json), but the builders apply the same full causal mask to all 24 layers, so sliding layers attend to the entire context.

Because prompts within 128 tokens make sliding attention equal to full attention, the short-prompt GPT-OSS E2E smoke (with lenient thresholds such as `logit_cosine_p5: 0.2`) keeps passing, which masked both defects.

## Environment

- GB300 (`galaxy-ts4-083-or`, aarch64), container `trtmc-dev-gb300`
- TensorRT 11.0.0.114, PyTorch 2.12.0+cu130, Transformers 5.2.0
- Model: `openai/gpt-oss-20b`, snapshot `6cee5e81ee83917806bbde320786a8fb61efebee`, fp16 engine, `gpt_oss_decoder_moe`
- Source run: MMLU five-shot task-eval diagnostics, 2026-07-06/07 (sample `mmlu_006604`, `--sample-seed 1234`, 408-token prompt)

## Affected Scope

- `python/tensorrt_model_connect/families/gpt_oss/plugin.py` (single-device MoE builder)
- `python/tensorrt_model_connect/families/gpt_oss/tp_builder.py` (TP builder)
- `python/tensorrt_model_connect/families/gpt_oss/graph_ops.py` (`make_yarn_rope_table_half_dim`)
- Any GPT-OSS engine serving prompts longer than 128 tokens (e.g. `mmlu_five_shot_mcq` task-eval, PR #396)

## Steps To Reproduce

```bash
/opt/venv/bin/python tools/task_eval.py eval \
  --suite mmlu_five_shot_mcq --model gpt-oss-20b \
  --dataset /mnt/data/MMLU_five_shot/mmlu_dataset.json \
  --single-device-only --limit 1 --sample-seed 1234 \
  --force-build --cuda-visible-devices 1 ...
```

Or feed the identical 408 tokenized prompt IDs to the family debug runner loaded from the built bundle (rules out the tokenizer).

## Actual Behavior

- TRT greedy output: token IDs `[2167, 30, 4066, 2167, 4066, 30, 4066, 1008]` → `"We?  \nWe  \n?  \n..."` (invalid, first generated token already wrong).
- HF reference on the same IDs: `[33, 200002]` → `"B<|return|>"` (valid MCQ answer).
- First-generation-step logit cosine HF vs TRT: 0.78; per-position prefill top-1 agreement 0.456 with divergence starting at position 0.

## Expected Behavior

The engine applies YaRN-scaled RoPE (HF-exact, including `truncate: false` and the attention factor) and masks sliding-attention layers to the configured 128-token window, matching `GptOssForCausalLM` up to fp16 noise.

## Root-Cause Evidence

Injecting exactly these two defects into the HF model (default RoPE with `attention_scaling=1` + all layers forced to full attention) reproduces the TRT output **token-for-token** (`[2167, 30, 4066, 2167, 4066, 30, 4066, 1008]`) with first-step logit cosine 0.9996 and 91.2% per-position top-1 agreement (the residual is bf16-vs-fp16 noise). Either defect alone does not match. This also rules out weights, MoE routing, attention sinks, cache handling, and the native tokenizer.

## Impact

Every GPT-OSS TRT engine is semantically wrong beyond 128 tokens of context; the nominal MMLU "pass" in earlier suite runs came from invalid/empty parse agreement, not correct behavior. CI does not catch it because the GPT-OSS smoke prompt is shorter than the sliding window and thresholds are lenient.

## Suspected Location

- `families/gpt_oss/plugin.py` `build_engine` rope-table block (reads `rope_parameters` only; no per-layer mask selection)
- `families/gpt_oss/tp_builder.py` `_make_rope_tables` / `build_gpt_oss_tp_engine` (same)
- `families/gpt_oss/graph_ops.py` `make_yarn_rope_table_half_dim` (always truncates; missing attention factor)

Fix branch with regression tests: `fix/gpt-oss-yarn-sliding-window` (PR to follow). Related: #396 (task-eval GPT-OSS contract that exposed this).
